diff --git a/potion.install b/potion.install
index b8610bd..dd0352f 100644
--- a/potion.install
+++ b/potion.install
@@ -117,7 +117,7 @@ function potion_requirements($phase) {
         'value'       => t('Missing libraries'),
         'severity'    => REQUIREMENT_ERROR,
         'description' => t('The following utilities are not found: <code>:utilities</code>. See <a href="https://www.drupal.org/docs/8/modules/potion/how-to-install-setup-gettext" target="_blank">How to install & setup Gettext</a> for more information.', [
-          ':utilities'   => implode($not_found, ', '),
+          ':utilities'   => implode(', ', $not_found),
         ]),
       ];
     }
@@ -128,7 +128,7 @@ function potion_requirements($phase) {
         'value'       => t('Not executable libraries'),
         'severity'    => REQUIREMENT_ERROR,
         'description' => t('The following utilities are not executable: <code>:utilities</code>. See <a href="https://www.drupal.org/docs/8/modules/potion/how-to-install-setup-gettext" target="_blank">How to install & setup Gettext</a> for more information.', [
-          ':utilities'   => implode($not_executable, ', '),
+          ':utilities'   => implode(', ', $not_executable),
         ]),
       ];
     }
