diff --git a/src/Plugin/openapi/OpenApiGeneratorBase.php b/src/Plugin/openapi/OpenApiGeneratorBase.php
index e4f0518..d18a9bf 100644
--- a/src/Plugin/openapi/OpenApiGeneratorBase.php
+++ b/src/Plugin/openapi/OpenApiGeneratorBase.php
@@ -261,26 +261,9 @@ abstract class OpenApiGeneratorBase extends PluginBase implements OpenApiGenerat
         case 'oauth2':
           $def = [
             'type' => 'oauth2',
-            'description' => 'For more information see https://developers.getbase.com/docs/rest/articles/oauth2/requests',
-            'flows' => [
-              'password' => [
-                'tokenUrl' => $base_url . 'oauth/token',
-                'refreshUrl' => $base_url . 'oauth/token',
-              ],
-              'authorizationCode' => [
-                'authorizationUrl' => $base_url . 'oauth/authorize',
-                'tokenUrl' => $base_url . 'oauth/token',
-                'refreshUrl' => $base_url . 'oauth/token',
-              ],
-              'implicit' => [
-                'authorizationUrl' => $base_url . 'oauth/authorize',
-                'refreshUrl' => $base_url . 'oauth/token',
-              ],
-              'clientCredentials' => [
-                'tokenUrl' => $base_url . 'oauth/token',
-                'refreshUrl' => $base_url . 'oauth/token',
-              ],
-            ],
+            'flow' => 'accessCode',
+            'authorizationUrl' => $base_url . 'oauth/authorize',
+            'tokenUrl' => $base_url . 'oauth/token',
           ];
           break;
         default:
